File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    var answer = 1;

    for(let i = 1; i <= n / 2; i++){
        let sum = i;
        for(let j = i + 1; j <= (n + 1) / 2; j++){
            sum += j;
            if(sum > n){
                break;
            }
            else if(sum == n){
                answer += 1;
                break;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;

        for(int i = 1; i <= n; i++){
            // 1부터 시작해서 쭈우우욱 더하기
            int count = i;
            int inc = i + 1;
            while(true){
                if(count == n) {answer ++; break;}
                else if(count > n) break;
                count += inc;
                inc++;
            }
        }
        
        return answer;
    }
}